api/envoy/extensions/filters/http/set_metadata/v3/set_metadata.proto

```diff
@@ -20,12 +22,51 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.filters.http.set_metadata]
 
 message Config {
+  message UntypedMetadata {
+    // The metadata namespace.
+    string metadata_namespace = 1 [(validate.rules).string = {min_len: 1}];
+
+    // The value to place at the namespace. If ``allow_overwrite``, this will
+    // overwrite or merge with any existing values in that namespace.
+    google.protobuf.Struct value = 2;
+
+    // Allow the filter to overwrite or merge with an existing value in the namespace.
+    bool allow_overwrite = 3;
+  }
+
+  message TypedMetadata {
+    // The metadata namespace.
+    string metadata_namespace = 1 [(validate.rules).string = {min_len: 1}];
+
+    // The value to place at the namespace. If ``allow_overwrite``, this will
+    // overwrite any existing values in that namespace. See
+    // :ref:`the filter documentation <config_http_filters_set_metadata>` for
+    // more information on how this value is merged with potentially existing
+    // ones if ``allow_overwrite`` is configured.
+    google.protobuf.Any value = 2;
+
+    // Allow the filter to overwrite an existing value in the namespace.
+    bool allow_overwrite = 3;
+  }
```

Comment:
IMO, it would be better to use one message that contains both value and typed_value to replace these two messages.

Reply:
Thanks for the suggestion. I have streamlined the API a bit